src/lib/components/modal/account/SaveNewAccount.tsx

```diff
@@ -109,6 +111,23 @@ export function SaveNewAccountModal({
     },
   });
 
+  const { refetch: refetchFull } = useAccountType(addressState, {
+    enabled: false,
+    onSuccess: (type) => {
+      if (type !== AccountType.ContractAccount) setStatus(statusSuccess);
+      else {
+        setErrorStatus("You need to save contract through Contract page.");
+        setIsContract(true);
+      }
+    },
+    onError: (err) => {
+      resetForm(false);
+      setErrorStatus(err.message);
+    },
+  });
```

Comment:
Please consider move logic inside `onSuccess` and `onError` of both `full` and `lite` version into separate function.